File: player.go
package main

const DefaultEnergy = 100
const DefaultMimimi = 0

type Player struct {
	Id         string `json:"id"`
	Name       string `json:"name"`
	Energy     int    `json:"energy"`
	Mimimi     int    `json:"mimimi"`
	Skin       string `json:"skin"`
	State      string `json:"state"`
	DiffEnergy int    `json:"diffEnergy"`
	DiffMimimi int    `json:"diffMimimi"`
}

func NewPlayer(socketId string, playerName string, skin string) (player *Player) {
	player = &Player{
		Id:         socketId,
		Name:       playerName,
		Skin:       skin,
		Mimimi:     DefaultMimimi,
		Energy:     DefaultEnergy,
		State:      "",
		DiffEnergy: 0,
		DiffMimimi: 0,
	}

	return
}

File: www/assets/js/game/player.js
'use strict';

class Player {
    constructor(phaserGame, id, name, skin, energy, mimimi, x, y) {
        this.id   = id;
        this.name = name;
        this.game = phaserGame;
        this.energy = energy;
        this.mimimi = mimimi;
        this.skin = skin;
        this.x = x;
        this.y = y;
        this.setState('stay');
    }

    setState(state) {
        if (this.sprite) {
            this.sprite.kill();
        }
        this.catStaySprite  = this.game.add.sprite(this.x, this.y, this.skin + '_' +  state);
        this.anim = this.catStaySprite.animations.add(this.skin + '_' + state);
        this.anim.play(10, true);
    }
}